crates/networking/p2p/sync.rs

```diff
@@ -0,0 +1,137 @@
+use std::sync::Arc;
+
+use ethrex_blockchain::error::ChainError;
+use ethrex_core::{
+    types::{Block, BlockHash, BlockHeader},
+    H256,
+};
+use ethrex_storage::Store;
+use tokio::{sync::Mutex, time::Instant};
+use tracing::{debug, info, warn};
+
+use crate::kademlia::KademliaTable;
+
+/// Manager in charge the sync process
+/// Only performs full-sync but will also be in charge of snap-sync in the future
+#[derive(Debug)]
+pub struct SyncManager {
+    // true: syncmode = snap, false = syncmode = full
+    #[allow(unused)]
+    snap_mode: bool,
+    peers: Arc<Mutex<KademliaTable>>,
+}
+
+impl SyncManager {
+    pub fn new(peers: Arc<Mutex<KademliaTable>>, snap_mode: bool) -> Self {
+        Self { snap_mode, peers }
+    }
+
+    /// Starts a sync cycle, updating the state with all blocks between the current head and the sync head
+    /// TODO: only uses full sync, should also process snap sync once implemented
+    pub async fn start_sync(&mut self, mut current_head: H256, sync_head: H256, store: Store) {
```

Comment:
so, if I understand this right, `current_head` is older than `sync_head`?

Reply:
yes

Reply:
I think eventually we will have to go backwards for syncing, but let's discuss it. Anyway this is great for a first version.